src/lib/RepositoryForms/Form/Processor/Content/UrlRedirectProcessor.php

```diff
@@ -120,15 +120,23 @@ private function resolveAdminSiteaccessRedirect(FormActionEvent $event): void
         /* If target URL was set to something else than Location, do nothing */
         try {
             $targetUrlAlias = $this->urlAliasService->lookup(
-                $response->getTargetUrl()
+                $response->getTargetUrl(),
+                null,
+                true
```

Comment:
The point here is that we shouldn't try to work around not working front-end route. This behavior of `lookup` is correct.

What you have here is exception-driven logic which is not a good approach. So while the choice to generate system route is correct, the place it is done seems like not the best choice.

In this case I think we should focus on changing the logic which generates this route as non-system one and sets it for the `$event` - so a few calls before.

So far I got to `\EzSystems\RepositoryForms\Form\Processor\ContentFormProcessor::processPublish` which generates that front-end route. We cannot change it there, because it also serves the purpose of UGC (probably), but extra data `redirectUrlAfterPublish` looks promising. That is of course if we're able to generate proper URL before this call.